test/support/response.rb

```diff
@@ -0,0 +1,88 @@
+require 'cgi'
+
+ERROR_IDS = %w(400 401 404 500)
+
+def index_response(resource, count)
+  {
+    :total_count => count,
+    :has_more => false,
+    :object => 'list',
+    :data => FactoryGirl.build_list(resource.to_sym, count)
+  }.to_json
+end
+
+def resource_from_uri(uri)
+  uri.to_s.split('/').last.singularize.to_sym
+end
+
+# Make sure the stubbed URI is valid.
+def check_uri_for_api_key(uri)
+  unless uri.user
+    fail ArgumentError, "URI doesn't include an API key"
+  end
+end
+
+def error_id?(id)
+  ERROR_IDS.include? id
+end
+
+def handle_test_response(request, id, action, factory_name)
+  if error_id? id
+    handle_error_response id
+  else
+    status = response_status request, action
+    body = response_body request, id, action, factory_name
+
+    build_test_response status, body
+  end
+end
+
+def response_status(request, action)
+  if request.method == :post && action.nil?
+    201
+  else
+    200
+  end
+end
+
+def response_body(request, id, action, factory_name)
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for response_body is too high. [15.56/15]